src/commands/runDraftTool/runDraftDeployment.ts

```diff
@@ -0,0 +1,665 @@
+import {Context} from './model/context';
+import * as vscode from 'vscode';
+import {State, StateApi} from '../../utils/state';
+import {longRunning} from '../../utils/host';
+import {downloadDraftBinary, runDraftCommand} from './helper/runDraftHelper';
+import {
+   AzureWizard,
+   AzureWizardExecuteStep,
+   AzureWizardPromptStep,
+   IActionContext
+} from '@microsoft/vscode-azext-utils';
+import {DraftFormat, draftFormats} from './model/format';
+import {ValidateImage, ValidateRfc1123} from '../../utils/validation';
+import {PromptPort, ignoreFocusOut} from './helper/commonPrompts';
+import {
+   KubernetesApi,
+   Kubernetes,
+   getDefaultKubeconfig,
+   getHelm,
+   getKubectl
+} from '../../utils/kubernetes';
+import {failed, getAysncResult} from '../../utils/errorable';
+import {
+   RegistryItem,
+   RepositoryItem,
+   ResourceGroupItem,
+   SubscriptionItem,
+   TagItem,
+   AzApi,
+   getAzureAccount,
+   Az
+} from '../../utils/az';
+import * as path from 'path';
+import * as fs from 'fs';
+import {
+   buildCreateCommand,
+   buildCreateConfig
+} from './helper/draftCommandBuilder';
+
+const title = 'Draft a Kubernetes Deployment and Service';
+
+enum imageOption {
+   ACR,
+   Other
+}
+interface PromptContext {
+   outputFolder: vscode.Uri;
+   applicationName: string;
+   format: DraftFormat;
+   namespace: string;
+   newNamespace: boolean;
+   image: string;
+   imageOption: imageOption;
+   acrSubscription: SubscriptionItem;
+   acrResourceGroup: ResourceGroupItem;
+   acrRegistry: RegistryItem;
+   acrRepository: RepositoryItem;
+   acrTag: TagItem;
+   port: string;
+}
+type WizardContext = IActionContext & Partial<PromptContext>;
+type IPromptStep = AzureWizardPromptStep<WizardContext>;
+type IExecuteStep = AzureWizardExecuteStep<WizardContext>;
+
+export async function runDraftDeployment(
+   {actionContext, extensionContext}: Context,
+   outputFolder: vscode.Uri | undefined
+) {
+   const state: StateApi = State.construct(extensionContext);
+   const kubeconfig = getDefaultKubeconfig();
+   const kubectlReturn = await getKubectl();
+   if (failed(kubectlReturn)) {
+      throw Error(kubectlReturn.error);
+   }
+   const helmReturn = await getHelm();
+   if (failed(helmReturn)) {
+      throw Error(helmReturn.error);
+   }
+   const k8s: KubernetesApi = new Kubernetes(
+      kubeconfig,
+      kubectlReturn.result,
+      helmReturn.result
```

Comment:
I may be missing something - but why do we need to keep track of helm here? Aren't we only using the k8s functionality to create the namespace?

Reply:
yes for now. The helm thing is for deploying helm later on. The Kubernetes class takes all three in. It's easier to build a full Kubernetes wrapper than handles everything rather than have separate ones or handle other logic.

The KubernetesApi doesn't specify that it needs helm or anything else so that part is just a particular implementation detail.

Reply:
makes sense!